src/components/SplitButton/SplitButton.tsx

```diff
@@ -1,69 +1,94 @@
-import useEventListener from "../../hooks/useEventListener";
 import DropdownChevronDown from "../Icon/Icons/components/DropdownChevronDown";
-
+import DialogContentContainer from "../DialogContentContainer/DialogContentContainer";
 // SCSS import
 import "./SplitButton.scss";
-import DialogContentContainer from "../DialogContentContainer/DialogContentContainer";
 
-// Constants
-const DIALOG_MOVE_BY = { main: 8, secondary: 0 };
-const DEFAULT_DIALOG_SHOW_TRIGGER = "click";
-const DEFAULT_DIALOG_HIDE_TRIGGER = ["clickoutside", "click", "esckey"];
-const SECONDARY_BUTTON_WRAPPER_CLASSNAME = "monday-style-split-button__secondary-button-wrapper";
-const EMPTY_ARR = [];
-
-const ENTER_KEYS = [keyCodes.ENTER];
-
-const SECONDARY_CONTENT_POSITIONS = {
-  BOTTOM_START: "bottom-start",
-  BOTTOM_MIDDLE: "bottom",
-  BOTTOM_END: "bottom-end"
-};
-
-export const SECONDARY_BUTTON_ARIA_LABEL = "additional actions";
-
-const SplitButton = ({
-  marginLeft,
-  marginRight,
-  success,
-  loading,
-  children,
-  leftIcon,
-  rightIcon,
-  color,
-  kind,
-  className,
-  onClick,
+export interface SplitButtonProps extends ButtonProps {
+  /*
+   * The element or renderer which display inside the dialog which open by clicking on the split button's secondary button.
+   */
+  secondaryDialogContent?: ReactElement | (() => void);
+  onSecondaryDialogDidShow?: () => void;
+  onSecondaryDialogDidHide?: () => void;
+  zIndex?: number;
+  /*
+   * Class name to provide the element which wraps the popover/modal/dialog
+   */
+  secondaryDialogClassName?: string;
+  secondaryDialogPosition?: SplitButtonSecondaryContentPosition;
+  /*
+    Popover Container padding size
+   */
+  // @ts-ignore TODO TS-migration, when DialogContentContainer is converted to TS
+  dialogPaddingSize?: typeof DialogContentContainer.sizes;
+  shouldCloseOnClickInsideDialog?: boolean;
+}
+
+const SplitButton: FC<SplitButtonProps> & {
+  secondaryPositions?: typeof SplitButtonSecondaryContentPosition;
+  secondaryDialogPositions?: typeof SplitButtonSecondaryContentPosition;
+  sizes?: typeof Button.sizes;
+  colors?: typeof Button.colors;
+  kinds?: typeof Button.kinds;
+  inputTags?: typeof Button.inputTags;
+  // @ts-ignore TODO TS-migration, when DialogContentContainer is converted to TS
+  dialogPaddingSizes?: typeof DialogContentContainer.sizes;
+} = ({
   secondaryDialogContent,
-  onSecondaryDialogDidShow,
-  onSecondaryDialogDidHide,
-  disabled,
+  onSecondaryDialogDidShow = NOOP,
+  onSecondaryDialogDidHide = NOOP,
   shouldCloseOnClickInsideDialog,
-  zIndex,
+  zIndex = null,
   secondaryDialogClassName,
-  secondaryDialogPosition,
-  dialogPaddingSize,
+  secondaryDialogPosition = SplitButtonSecondaryContentPosition.BOTTOM_START,
+  // @ts-ignore TODO TS-migration, when DialogContentContainer is converted to TS
+  dialogPaddingSize = DialogContentContainer.sizes.MEDIUM,
   ...buttonProps
 }) => {
+  const overrideButtonProps = { ...Button.defaultProps, ...buttonProps };
```

Comment:
Why we need Button.defaultProps? when we pass it to button, the button default props will override undefined values?

Reply:
Yes, but these button props are used to calculate props and classes for SplitButton as well - not only for Button. That's why we have to know their values before passing them to Button